GoogleForCreators/web-stories-wp
# Try re-running failed karma tests once

## Summary

1. Collect names of all failed specs, store them in a temporary file
1. Use regex to only run these specs in a second run

## Relevant Technical Choices

`specFilter` as suggested in https://github.com/google/web-stories-wp/issues/2792#issuecomment-658467793 is not usable as karma-jasmine sets this one itself. But `grep` seemed to do the trick.

## To-do

N/A

## User-facing changes

N/A

## Testing Instructions

1. Modify one test so it fails.
1. Run `npm run test:karma:edit-story -- --headless --snapshots --viewport=1600:1000 --coverage`
1. One test should fail
1. Run `npm run test:karma:edit-story:retry-failed -- --headless --snapshots --viewport=1600:1000`
1. Only the failed test should fail

---



Fixes #2792
